scipy/interpolate/_bspl.pyx

```diff
@@ -457,9 +459,13 @@ def _make_design_matrix(const double[::1] x,
         ind = find_interval(t, k, xval, ind, extrapolate)
         _deBoor_D(&t[0], xval, k, ind, 0, &work[0])
 
-        data[(k + 1) * i : (k + 1) * (i + 1)] = work[:k + 1]
-        row_ind[(k + 1) * i : (k + 1) * (i + 1)] = i
-        col_ind[(k + 1) * i : (k + 1) * (i + 1)] = np.arange(
-            ind - k, ind + 1, dtype=int)
+        # data[(k + 1) * i : (k + 1) * (i + 1)] = work[:k + 1]
+        # indices[(k + 1) * i : (k + 1) * (i + 1)] = np.arange(ind - k, ind + 1)
+        for j in range((k + 1)):
+            data[(k + 1) * i + j] = work[j]
+            indices[(k + 1) * i + j] = ind - k + j
 
-    return np.asarray(data), (row_ind, col_ind)
+    return csr_array(
+        (np.asarray(data), np.asarray(indices), np.asarray(indptr)),
+        shape=(n, t.shape[0] - k - 1),
+    )
```

Comment:
Let's keep importing from scipy.sparse in `_bsplines.py` only and keep this cython function returning arrays to be wrapped into a `csr_constructor`, if only for better visibility of the sparse-interpolate dependence. 

Would be nice to check the binary size with and without `from scipy.sparse import csr_array`, but that's not a requirement. Quite likely that there is no effect. But "not hiding the subpackage interdependence" is I think a good reason, unless there is a serious performance or other benefit.

Reply:
I don't follow that logic. Cython code always mixes with Python, that's the whole point. Having the aimed for output, i.e. a csr arrya, is a much nicer API, IMHO.